tasks/Task.cpp

```diff
@@ -12,44 +12,89 @@
 #include <iostream>
 #include <json/json.h>
 
+#include <list>
+
 using namespace controldev_websocket;
 using namespace controldev;
 using namespace seasocks;
 
-struct controldev_websocket::JoystickHandler : WebSocket::Handler {
+struct controldev_websocket::Client {
     WebSocket *connection = nullptr;
+    std::string id = "";
+    Json::FastWriter fast;
+    bool is_new = true;
+
+    bool isEmpty(){
+        return connection == nullptr;
+    }
+
+    void steal(Client &other){
+        if (other.isEmpty() and !is_new){
+            return;
+        }
+        Json::Value feedback;
+        if (other.isEmpty()){
+            feedback["connection_state"]["state"] = "new";
+        } else {
+            feedback["connection_state"]["state"] = "lost";
+            feedback["connection_state"]["peer"] = id.empty() ? "pending connection" : id;
+            other.connection->send(fast.write(feedback));
+            other.connection->close();
+            feedback["connection_state"]["state"] = "stolen";
+            feedback["connection_state"]["peer"] = other.id.empty() ? "pending connection"
+                                                                      : other.id;
+        }
+        connection->send(fast.write(feedback));
+        is_new = false;
+    };
+};
+struct controldev_websocket::JoystickHandler : WebSocket::Handler {
+    Client pending;
+    Client controlling;
     Task *task = nullptr;
     Json::Value msg;
     Json::FastWriter fast;
     Statistics statistic;
 
     void onConnect(WebSocket *socket) override{
-        if (connection != nullptr){
-            connection->close();
-        }
-        connection = socket;
-        task->is_controlling = false;
+        Client new_pending;
+        new_pending.connection = socket;
+        new_pending.steal(pending);
```

Comment:
Don't call `steal` if you don't mean to `steal`. A.k.a. the `steal` method name is terrible since it's not always stealing anything.

Reply:
I think you need to separate the "stealing pending" and "stealing controlling", or have a "stage" field that says whether the feedback is about the pending connection establishment, or controlling connection establishment.

Reply:
> I think you need to separate the "stealing pending" and "stealing controlling", or have a "stage" field that says whether the feedback is about the pending connection establishment, or controlling connection establishment.

Well, if the stealing from pending the peer will be "pending connection"... Isn't it enough? (I just get the id after the handshake, so before the id is empty maybe it should be explicit at code)

Reply:
> Don't call `steal` if you don't mean to `steal`. A.k.a. the `steal` method name is terrible since it's not always stealing anything.

I don't understand, the action is to steal. You try to steal but is empty, but from my perspective, you are always a thief. If the connection was a bank and you call steal function would say that steal is wrong because the bank may be empty? Anyways, what do you suggest? `takeConnection`, `tryToSteal`?